### PR TITLE
refactor item queue

### PR DESCRIPTION
**Pull request by rlauuzo**

### CODE REVIEW

**Comment by rlauuzo** [comment]
- Adds blacklist https://github.com/jumoog/intro-skipper/pull/177#issuecomment-2123215908
- Removes path restriction from automatic tasks and passes item IDs to the analyzer instead.

**Comment by AbandonedCart** [comment]
The path restrictions was a super hacky way about it. Glad to see it go.

**Comment by jumoog** [comment]
so we can remove PathRestrictions from the config?

**Comment by AbandonedCart** [comment]
It’s still being used right now, but this PR will make it obsolete

**Comment by jumoog** [comment]
Yes, that is my point, if it is obsolete we should remove it from the configuration page to avoid confusion.

**Comment by AbandonedCart** [comment]
It’s not in the configuration page. It’s a silent option.

**Comment by jumoog** [comment]
Oh yes, I just checked the code. Sorry for the confusion.

File: ConfusedPolarBear.Plugin.IntroSkipper/Entrypoint.cs [hunk 1]
```diff
@@ -149,7 +149,7 @@ private void OnItemAdded(object? sender, ItemChangeEventArgs itemChangeEventArgs
             return;
         }
 
-        _episodeIdsToAnalyze.Add(episode.Id);
+        _seasonsToAnalyze.Add(episode.SeasonId);
```
Comment (AbandonedCart) [inline ConfusedPolarBear.Plugin.IntroSkipper/Entrypoint.cs:152]: I may have missed where you already do it, but you may want to check that it's not already added for add and edit. As far as I know, they fire for each file.

**Comment by AbandonedCart** [comment]
I guess that confirms a question i've always had in the back of my mind. Does anime even have the black frames?

**Comment by rlauuzo** [comment]
Animes on Netflix often have a standard anime outro followed by black frames displaying credits for all the languages and dubs.

**Comment by AbandonedCart** [comment]
The dreaded *originals* always have follow-up credits.

**Comment by AbandonedCart** [comment]
I think this one should be prioritized. A lot of the changes here will drastically improve performance and will also open a lot of potential for improving the automatic scans even further.

**Comment by jumoog** [comment]
whats a good anime for testing?

**Comment by AbandonedCart** [comment]
My go to has been Demon Slayer because the first episode of the season always puts the intro at the end . It has an intro, credits, and preview. The intro ends in a splash screen so it's easy to locate.

**Review by jumoog — APPROVED** [review]
tests are still working and code looks good.

**Comment by Gauvino** [comment]
Don't know about the code but been running this PR for 2 days without any problem

**Comment by jumoog** [comment]
is this ready?

**Comment by rlauuzo** [comment]
I guess so. I'm not entirely sure what the optimal way is to handle the lists for loading existing fingerprints and other data in the Chromaprint analyzer. That being said, the Chromaprint analyzer has some other areas that could be improved, like the silence detection, which seems a bit pointless in its current implementation. But those refinements are probably best left for another time.

**Comment by jumoog** [comment]
Yes, I think we can merge it in this form for now and make a new release.

**Pull request merged**